src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -939,7 +939,7 @@ void ControlBoardWrapper::run()
 //
 bool ControlBoardWrapper::setPid(int j, const Pid &p)
 {
-    int off=device.lut[j].offset;
+    int off; try{off = device.lut.at(j).offset;} catch(...){yError() << "joint number " << j <<  " out of bound [0-"<< controlledJoints << "] for part " << partName; return false; }
```

Comment:
But the `vector::at` method is implemented with two `if`, so I fail to see the efficiency gain coming from using the exception.  
Furthermore for this kind of behavior profiling is the only way of actually knowing the speed of the code, and typically results are quite counterintuitive.